opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -95,14 +95,103 @@ message InstrumentationLibraryMetrics {
+// - StartTimeUnixNano is optional, but strongly encouraged for DataPoints
+//   having an AggregationTemporality field, as discussed below.
+//
+// Both TimeUnixNano and StartTimeUnixNano values are expressed as
+// UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+//
+// # TimeUnixNano
+//
+// This field is required, having consistent interpretation across
+// DataPoint types.  TimeUnixNano is the moment corresponding to when
+// the data point's aggregate value was captured.
+//
+// Data points with/ the 0 value for TimeUnixNano SHOULD be rejected
+// by consumers.
+//
+// # StartTimeUnixNano
+//
+// This field indicates to consumers the start time for points with
+// cumulative and delta AggregationTemporality, and it should be
+// included whenever possible to support correct rate calculation and
+// overlap detection.  Although it may be omitted when the start time
+// is truly unknown, setting StartTimeUnixNano is strongly encouraged.
+//
+// There are two reasons that streams with AggregationTemporality are
+// reset: (1) a process restarts, and (2) the SDK "forgets" the stream
+// and then re-initializes it.  StartTimeUnixNano is critical for
+// calculating correct rates for Monotonic Sum, Histogram, and Summary
+// data points in both of these cases
+//
+// StartTimeUnixNano may be used with Gauge instruments to indicate
+// when the measurement was captured, as opposed to the moment it was
+// observed.
+//
+// Another use for this information is to prevent overlap when
+// multiple writers accidentally write to a single stream--applies
+// to all data points.
+//
+// # Unknown StartTimeUnixNano
+//
+// This protocol adopts the Prometheus heuristic for reset detection,
+// and with it prescribes a solution for filling in StartTimeUnixNanos
+// with as much information as possible to assist consumers with
+// calculating rates and detecting overlap.
+//
+// The Prometheus heuristic indicates that a reset must have occured
+// in a monotonic series whenever the value descends from one value to
+// the next, in a timestamp-ordered sequence.
+//
+// To facilitate reset and overlap detection, when single-timestamped
+// sample values are translated into OTLP, such as from the Prometheus
+// Remote-Write protocol, use the following.
+//
+// ## Filling in StartTimeUnixNano
+//
+// To import a sequence of non-overlapping measurements into into OTLP
+// using a stateful processor:
+//
+// 1. Keep a map, using the metric name and attributes as the key, containing
+//    a struct of three parts:
+//    - resetTimestamp: the first observation or the reset timestamp
+//    - previousTimestamp: the timestamp of the most recent observation
+//    - previousValue: a monotonic value
+// 2. When a new point arrives and it is the first observation:
+//    - in the map, set resetTimestamp to the input timestamp
+//    - in the map, set previousValue to the input value
+//    - in the map, set previousTimestamp to resetTimestamp
```

Comment:
Does this say that previousTimestamp is set to the input timestamp? Unclear why use a different wording.